linters/import/main.go

```diff
@@ -0,0 +1,438 @@
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package main
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"go/ast"
+	"go/build"
+	"go/parser"
+	"go/token"
+	"go/types"
+	"log"
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/kisielk/gotool"
+	"golang.org/x/tools/go/loader"
+)
+
+const (
+	standard = "standard"
+	external = "ext"
+)
+
+var (
+	errMultipleImport       = errors.New("more than one import group found")
+	errOutOfOrder           = errors.New("import is out of order")
+	errImportMatchedAlready = errors.New("import already matched pattern")
+	errGroupMachedAlready   = errors.New("import group already matches previously seen pattern")
+	errNoMatch              = errors.New("import does not match any of the provided patterns")
+	errTooManyImports       = errors.New("number of import groups exceeds number of patterns. this usually means a previous or future import is out of order or you didn't specify the correct patterns")
+)
+
+type lintError struct {
+	fileName    string
+	importName  string
+	line        int
+	patternSeen string
+	err         error
+}
+
+type lintErrors []lintError
+
+// NB(prateek): http://goast.yuroyoro.net/ is enormously helpful.
+
+func main() {
+	tags := flag.String("tags", "", "List of build tags to take into account when linting.")
+	skipVendor := flag.Bool("skip-vendor", true, "Skip vendor directors.")
+	rawPatterns := flag.String("patterns", "", "Specify the patterns of each group in order. If checking for Go standard imports write `standard`, if checking for a wildard group write `all`.")
+
+	flag.Parse()
+	importPaths := gotool.ImportPaths(flag.Args())
+	if len(importPaths) == 0 {
+		flag.Usage()
+		return
+	}
+
+	patterns := strings.Fields(*rawPatterns)
+	if len(patterns) < 1 {
+		fmt.Fprint(os.Stderr, "List of patterns must be greater than 0\n")
+		os.Exit(1)
+	}
+
+	var filteredPaths []string
+	if *skipVendor {
+		filteredPaths = filterOutVendor(importPaths)
+	} else {
+		filteredPaths = importPaths
+	}
+
+	groupedErrors := handleImportPaths(filteredPaths, strings.Fields(*tags), patterns)
+	printErrors(groupedErrors)
+}
+
+func printErrors(groupedErrors []lintErrors) {
+	if len(groupedErrors) != 0 {
+		fmt.Printf("Number of import ordering issues found: %d\n\n", len(groupedErrors))
+		for _, group := range groupedErrors {
+			for _, imp := range group {
+				fmt.Printf("File: %s\nImport: %s\nLine: %d\nPattern: %s\nError: %v\n\n", imp.fileName, imp.importName, imp.line, imp.patternSeen, imp.err)
```

Comment:
could you ensure you're printing is compatible with how the metalinter parses output (e.g. https://github.com/m3db/build-tools/blob/master/linters/badtime/main.go#L211 is how the other linter prints issues)